src/compiler/checker.ts

```diff
@@ -6455,6 +6456,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     function createNodeBuilder() {
         return {
             typeToTypeNode: (type: Type, enclosingDeclaration?: Node, flags?: NodeBuilderFlags, tracker?: SymbolTracker) => withContext(enclosingDeclaration, flags, tracker, context => typeToTypeNodeHelper(type, context)),
+            expressionOrTypeToTypeNode: (type: Type, expr: Expression | JsxAttributeValue | undefined, addUndefined?: boolean, enclosingDeclaration?: Node, flags?: NodeBuilderFlags, tracker?: SymbolTracker) => withContext(enclosingDeclaration, flags, tracker, context => expressionOrTypeToTypeNode(context, expr, type, addUndefined)),

```

Comment:
The ordering of these parameters is a little strange. Should `expr` go first?

Reply:
`expressionOrType` matches the priority of the generator, but the argument order matches the required-ness of the types (`type` being required, and `expr` being allowed to be `undefined`). I can swap the arguments to match the name if you think that consistency is more important.

Reply:
It's not really meant to be an optional argument, just a potentially undefined one - so I'd prefer the `expr, type` ordering.